InternetHospitalWebUI/src/app/Services/authentication.service.ts

```diff
@@ -32,12 +33,27 @@ export class  AuthenticationService  {
                     //set flag that a user is logged in
                     this.isLoginSubject.next(true);
                     //set user role depending on the token claims
-                    this.setUserRole()                   
+                    this.setUserRole();
+                    this.AvatarURL.next(this.url + user.user_avatar);                    
```

Comment:
You have checkAvatarUrl() method, so use it in this.AvatarURL.next(...). Without it there are gonna be some bugs if user doesn't have avatar.